Sage/carbon
# ci: remove renogen

### Proposed behaviour
BREAKING CHANGE: This removes renogen in favour of semantic-release
This includes all of the outstanding un-released functionality that was merged with renogen notes.

### Current behaviour
We use renogen to generate the release notes.

### Checklist

<del>- [ ] Unit tests</del>
<del>- [ ] Cypress automation tests</del>
<del>- [ ] Storybook added or updated</del>
<del>- [ ] Theme support</del>
<del>- [ ] Typescript `d.ts` file added or updated</del>

### Additional context
#2549

### Testing instructions
You can see the contents of the release notes in the PR, when merged semantic release will trigger a release due to the the `BREAKING CHANGE` commit message. Unfortunately the semantic changelog will not include the release notes that were previously written in renogen. I'll manually add them to the github release.
